drush/civicrm.drush.inc

```diff
@@ -512,6 +513,10 @@ function _civicrm_generate_settings_file($dbuser, $dbpass, $dbhost, $dbname, $mo
     $cms = $v['cms'];
   }
 
+  if (!$sitekey = drush_get_option('site_key', FALSE)) {
+      $sitekey = md5(uniqid('', TRUE) . $baseUrl);
```

Comment:
This is tangential, but reading this code reminds me... `uniqid()` function is not a great source of entropy. I think this snippet of code probably originated in a much earlier iteration of `civicrm-core:install/civicrm.php`. That was a changed a while back (*a few years ago?*), so I'd suggest copying [a more recent snippet](https://github.com/civicrm/civicrm-core/blob/5.16/install/civicrm.php#L292), i.e.

```suggestion
      $sitekey = md5(rand() . mt_rand() . rand() . uniqid('', TRUE) . $baseUrl);
```

(*Long term, we should swap this over to `civicrm-setup` so that code is not replicated as much; and we should update the shared code to prioritize better sources of entropy when available. But that's a bigger scope. The suggestion here is just sync-up with the current `install/civicrm.php`.*)

Reply:
yes I agree longer term this would be good to reuse code rather than replicate it, I'd noted that when I was looking at some other issues a few weeks ago (it puzzled me slightly).  I'll add the code snippet you suggested and fix the few code style issues.  The function I used was the one already in the drush script.  I think an overhaul of this file would be useful at some stage, it's quite messy and will be difficult to maintain.

Reply:
I've added the new sitekey gen function (copied from civicrm_setup as suggested) - I should have commited the suggestion above but didn't see that button until afterwards.

I've run tests without the parameter, with the parameter but no value and with the parameter and a value and they all work as expected.